File: NegocioCSharp/EmailSalida.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;

namespace LiquidacionSueldos
{
    namespace Negocio
    {
        public partial class EmailSalida
        {

            private Datos.Gestor ocdGestor = new Datos.Gestor();
            private DataTable Fila;

            private DataTable Tabla = new DataTable();
            private int _esaId;
            public int esaId
            {
                get { return _esaId; }
                set { _esaId = value; }
            }

            private System.DateTime _esaFecha;
            public System.DateTime esaFecha
            {
                get { return _esaFecha; }
                set { _esaFecha = value; }
            }

            private string _esaPara;
            public string esaPara
            {
                get { return _esaPara; }
                set { _esaPara = value; }
            }

            private string _esaTipo;
            public string esaTipo
            {
                get { return _esaTipo; }
                set { _esaTipo = value; }
            }

            private string _esaTitulo;
            public string esaTitulo
            {
                get { return _esaTitulo; }
                set { _esaTitulo = value; }
            }

            private string _esaCuerpo;
            public string esaCuerpo
            {
                get { return _esaCuerpo; }
                set { _esaCuerpo = value; }
            }

            private string _esaDescripcion;
            public string esaDescripcion
            {
                get { return _esaDescripcion; }
                set { _esaDescripcion = value; }
            }

            private bool _esaActivo;
            public bool esaActivo
            {
                get { return _esaActivo; }
                set { _esaActivo = value; }
            }

            private System.DateTime _esaFechaHoraCreacion;
            public System.DateTime esaFechaHoraCreacion
            {
                get { return _esaFechaHoraCreacion; }
                set { _esaFechaHoraCreacion = value; }
            }

            private System.DateTime _esaFechaHoraUltimaModificacion;
            public System.DateTime esaFechaHoraUltimaModificacion
            {
                get { return _esaFechaHoraUltimaModificacion; }
                set { _esaFechaHoraUltimaModificacion = value; }
            }

            private int _cuoId;
            public int cuoId
            {
                get { return _cuoId; }
                set { _cuoId = value; }
            }

            private int _usuIdCreacion;
            public int usuIdCreacion
            {
                get { return _usuIdCreacion; }
                set { _usuIdCreacion = value; }
            }

            private int _usuIdUltimaModificacion;
            public int usuIdUltimaModificacion
            {
                get { return _usuIdUltimaModificacion; }
                set { _usuIdUltimaModificacion = value; }
            }

            public EmailSalida()
            {
                try
                {
                    this.esaId = 0;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public EmailSalida(int esaId)
            {
                try
                {
                    Fila = new DataTable();
                    Fila = ocdGestor.EjecutarReader("[EmailSalida.ObtenerUno]", new object[,] { {
                        "@esaId",
                        esaId
                    } });

                    if (Fila.Rows.Count > 0)
                    {
                        if (Fila.Rows[0]["esaId"].ToString().Trim().Length > 0)
                        {
                            this.esaId = Convert.ToInt32(Fila.Rows[0]["esaId"]);
                        }
                        else
                        {
                            this.esaId = 0;
                        }

                        if (Fila.Rows[0]["esaFecha"].ToString().Trim().Length > 0)
                        {
                            this.esaFecha = Convert.ToDateTime(Fila.Rows[0]["esaFecha"]);
                        }
                        else
                        {
                            this.esaFecha = System.DateTime.Now;
                        }

                        if (Fila.Rows[0]["esaPara"].ToString().Trim().Length > 0)
                        {
                            this.esaPara = Convert.ToString(Fila.Rows[0]["esaPara"]);
                        }
                        else
                        {
                            this.esaPara = "";
                        }

                        if (Fila.Rows[0]["esaTipo"].ToString().Trim().Length > 0)
                        {
                            this.esaTipo = Convert.ToString(Fila.Rows[0]["esaTipo"]);
                        }
                        else
                        {
                            this.esaTipo = "";
                        }

                        if (Fila.Rows[0]["esaTitulo"].ToString().Trim().Length > 0)
                        {
                            this.esaTitulo = Convert.ToString(Fila.Rows[0]["esaTitulo"]);
                        }
                        else
                        {
                            this.esaTitulo = "";
                        }

                        if (Fila.Rows[0]["esaCuerpo"].ToString().Trim().Length > 0)
                        {
                            this.esaCuerpo = Convert.ToString(Fila.Rows[0]["esaCuerpo"]);
                        }
                        else
                        {
                            this.esaCuerpo = "";
                        }

                        if (Fila.Rows[0]["esaDescripcion"].ToString().Trim().Length > 0)
                        {
                            this.esaDescripcion = Convert.ToString(Fila.Rows[0]["esaDescripcion"]);
                        }
                        else
                        {
                            this.esaDescripcion = "";
                        }

                        if (Fila.Rows[0]["esaFechaHoraCreacion"].ToString().Trim().Length > 0)
                        {
                            this.esaFechaHoraCreacion = Convert.ToDateTime(Fila.Rows[0]["esaFechaHoraCreacion"]);
                        }
                        else
                        {
                            this.esaFechaHoraCreacion = System.DateTime.Now;
                        }

                        if (Fila.Rows[0]["esaFechaHoraUltimaModificacion"].ToString().Trim().Length > 0)
                        {
                            this.esaFechaHoraUltimaModificacion = Convert.ToDateTime(Fila.Rows[0]["esaFechaHoraUltimaModificacion"]);
                        }
                        else
                        {
                            this.esaFechaHoraUltimaModificacion = System.DateTime.Now;
                        }

                        if (Fila.Rows[0]["esaActivo"].ToString().Trim().Length > 0)
                        {
                            this.esaActivo = (Convert.ToInt32(Fila.Rows[0]["esaActivo"]) == 1 ? true : false);
                        }
                        else
                        {
                            this.esaActivo = false;
                        }

                        if (Fila.Rows[0]["cuoId"].ToString().Trim().Length > 0)
                        {
                            this.cuoId = Convert.ToInt32(Fila.Rows[0]["cuoId"]);
                        }
                        else
                        {
                            this.cuoId = 0;
                        }

                        if (Fila.Rows[0]["usuIdCreacion"].ToString().Trim().Length > 0)
                        {
                            this.usuIdCreacion = Convert.ToInt32(Fila.Rows[0]["usuIdCreacion"]);
                        }
                        else
                        {
                            this.usuIdCreacion = 0;
                        }

                        if (Fila.Rows[0]["usuIdUltimaModificacion"].ToString().Trim().Length > 0)
                        {
                            this.usuIdUltimaModificacion = Convert.ToInt32(Fila.Rows[0]["usuIdUltimaModificacion"]);
                        }
                        else
                        {
                            this.usuIdUltimaModificacion = 0;
                        }

                    }
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public EmailSalida(int esaId, System.DateTime esaFecha, string esaPara, string esaTipo, string esaTitulo, string esaCuerpo, string esaDescripcion, bool esaActivo, System.DateTime esaFechaHoraCreacion, System.DateTime esaFechaHoraUltimaModificacion,
            int IcuoId, int IusuIdCreacion, int IusuIdUltimaModificacion)
            {
                try
                {
                    this.esaId = esaId;
                    this.esaFecha = esaFecha;
                    this.esaPara = esaPara;
                    this.esaTipo = esaTipo;
                    this.esaTitulo = esaTitulo;
                    this.esaCuerpo = esaCuerpo;
                    this.esaDescripcion = esaDescripcion;
                    this.esaActivo = esaActivo;
                    this.esaFechaHoraCreacion = esaFechaHoraCreacion;
                    this.esaFechaHoraUltimaModificacion = esaFechaHoraUltimaModificacion;
                    this.cuoId = cuoId;
                    this.usuIdCreacion = usuIdCreacion;
                    this.usuIdUltimaModificacion = usuIdUltimaModificacion;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }


            public DataTable ObtenerParametros()
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[EmailSalida.ObtenerParametros]", new object[,]{});
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public DataTable ObtenerTodo()
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[EmailSalida.ObtenerTodo]", new object[,]{});
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public DataTable ObtenerTodoCustom()
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[EmailSalida.ObtenerTodoCustom]", new object[,]{});
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public DataTable ObtenerUno(int esaId)
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[EmailSalida.ObtenerUno]", new object[,] { {
                        "@esaId",
                        esaId
                    } });
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public DataTable ObtenerValidarParametro()
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[EmailSalida.ObtenerValidarParametro]", new object[,]{});
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public void Actualizar(int esaId, System.DateTime esaFecha, string esaPara, string esaTipo, string esaTitulo, string esaCuerpo, string esaDescripcion, int cuoId, bool esaActivo, int usuIdCreacion,
            int usuIdUltimaModificacion, System.DateTime esaFechaHoraCreacion, System.DateTime esaFechaHoraUltimaModificacion)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[EmailSalida.Actualizar]", new object[,] {
                        {
                            "@esaId",
                            esaId
                        },
                        {
                            "@esaFecha",
                            esaFecha
                        },
                        {
                            "@esaPara",
                            esaPara
                        },
                        {
                            "@esaTipo",
                            esaTipo
                        },
                        {
                            "@esaTitulo",
                            esaTitulo
                        },
                        {
                            "@esaCuerpo",
                            esaCuerpo
                        },
                        {
                            "@esaDescripcion",
                            esaDescripcion
                        },
                        {
                            "@cuoId",
                            cuoId
                        },
                        {
                            "@esaActivo",
                            esaActivo
                        },
                        {
                            "@usuIdCreacion",
                            usuIdCreacion
                        },
                        {
                            "@usuIdUltimaModificacion",
                            usuIdUltimaModificacion
                        },
                        {
                            "@esaFechaHoraCreacion",
                            esaFechaHoraCreacion
                        },
                        {
                            "@esaFechaHoraUltimaModificacion",
                            esaFechaHoraUltimaModificacion
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Eliminar(int esaId)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[EmailSalida.Eliminar]", new object[,] { {
                        "@esaId",
                        esaId
                    } });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Generar(string QueEnviar, int Mes, int Año, int usuId)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[EmailSalida.Generar]", new object[,] {
                        {
                            "@QueEnviar",
                            QueEnviar
                        },
                        {
                            "@Mes",
                            Mes
                        },
                        {
                            "@Año",
                            Año
                        },
                        {
                            "@usuId",
                            usuId
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Insertar(System.DateTime esaFecha, string esaPara, string esaTipo, string esaTitulo, string esaCuerpo, string esaDescripcion, int cuoId, bool esaActivo, int usuIdCreacion, int usuIdUltimaModificacion,
            System.DateTime esaFechaHoraCreacion, System.DateTime esaFechaHoraUltimaModificacion)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[EmailSalida.Insertar]", new object[,] {
                        {
                            "@esaFecha",
                            esaFecha
                        },
                        {
                            "@esaPara",
                            esaPara
                        },
                        {
                            "@esaTipo",
                            esaTipo
                        },
                        {
                            "@esaTitulo",
                            esaTitulo
                        },
                        {
                            "@esaCuerpo",
                            esaCuerpo
                        },
                        {
                            "@esaDescripcion",
                            esaDescripcion
                        },
                        {
                            "@cuoId",
                            cuoId
                        },
                        {
                            "@esaActivo",
                            esaActivo
                        },
                        {
                            "@usuIdCreacion",
                            usuIdCreacion
                        },
                        {
                            "@usuIdUltimaModificacion",
                            usuIdUltimaModificacion
                        },
                        {
                            "@esaFechaHoraCreacion",
                            esaFechaHoraCreacion
                        },
                        {
                            "@esaFechaHoraUltimaModificacion",
                            esaFechaHoraUltimaModificacion
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void RegistrarEnvio(int esaId)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[EmailSalida.RegistrarEnvio]", new object[,] { {
                        "@esaId",
                        esaId
                    } });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Actualizar()
            {
                try
                {
                    if (this.esaId != 0)
                    {
                        ocdGestor.EjecutarNonQuery("[EmailSalida.Actualizar]", new object[,] {
                            {
                                "@esaId",
                                esaId
                            },
                            {
                                "@esaFecha",
                                esaFecha
                            },
                            {
                                "@esaPara",
                                esaPara
                            },
                            {
                                "@esaTipo",
                                esaTipo
                            },
                            {
                                "@esaTitulo",
                                esaTitulo
                            },
                            {
                                "@esaCuerpo",
                                esaCuerpo
                            },
                            {
                                "@esaDescripcion",
                                esaDescripcion
                            },
                            {
                                "@cuoId",
                                cuoId
                            },
                            {
                                "@esaActivo",
                                esaActivo
                            },
                            {
                                "@usuIdCreacion",
                                usuIdCreacion
                            },
                            {
                                "@usuIdUltimaModificacion",
                                usuIdUltimaModificacion
                            },
                            {
                                "@esaFechaHoraCreacion",
                                esaFechaHoraCreacion
                            },
                            {
                                "@esaFechaHoraUltimaModificacion",
                                esaFechaHoraUltimaModificacion
                            }
                        });
                    }
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Eliminar()
            {
                try
                {
                    if (this.esaId != 0)
                    {
                        ocdGestor.EjecutarNonQuery("[EmailSalida.Eliminar]", new object[,] { {
                            "@esaId",
                            esaId
                        } });
                    }
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public int Insertar()
            {
                int IdMax = 0;
                try
                {
                    IdMax = ocdGestor.EjecutarNonQueryRetornaId("[EmailSalida.Insertar]", new object[,] {
                        {
                            "@esaFecha",
                            esaFecha
                        },
                        {
                            "@esaPara",
                            esaPara
                        },
                        {
                            "@esaTipo",
                            esaTipo
                        },
                        {
                            "@esaTitulo",
                            esaTitulo
                        },
                        {
                            "@esaCuerpo",
                            esaCuerpo
                        },
                        {
                            "@esaDescripcion",
                            esaDescripcion
                        },
                        {
                            "@cuoId",
                            cuoId
                        },
                        {
                            "@esaActivo",
                            esaActivo
                        },
                        {
                            "@usuIdCreacion",
                            usuIdCreacion
                        },
                        {
                            "@usuIdUltimaModificacion",
                            usuIdUltimaModificacion
                        },
                        {
                            "@esaFechaHoraCreacion",
                            esaFechaHoraCreacion
                        },
                        {
                            "@esaFechaHoraUltimaModificacion",
                            esaFechaHoraUltimaModificacion
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
                return IdMax;
            }

        }
    }
}

File: Negocio/EmailSalida.vb
Namespace LiquidacionSueldos
    Namespace Negocio
        Partial Public Class EmailSalida

            Private ocdGestor As New LiquidacionSueldos.Datos.Gestor
            Private Fila As DataTable
            Private Tabla As New DataTable

			Private _esaId As Integer
            Public Property esaId() As Integer
                Get
					Return _esaId
                End Get
                Set(ByVal value As Integer)
                    _esaId = value
                End Set
            End Property

			Private _esaFecha As Date
            Public Property esaFecha() As Date
                Get
					Return _esaFecha
                End Get
                Set(ByVal value As Date)
                    _esaFecha = value
                End Set
            End Property

			Private _esaPara As String
            Public Property esaPara() As String
                Get
					Return _esaPara
                End Get
                Set(ByVal value As String)
                    _esaPara = value
                End Set
            End Property

			Private _esaTipo As String
            Public Property esaTipo() As String
                Get
					Return _esaTipo
                End Get
                Set(ByVal value As String)
                    _esaTipo = value
                End Set
            End Property

			Private _esaTitulo As String
            Public Property esaTitulo() As String
                Get
					Return _esaTitulo
                End Get
                Set(ByVal value As String)
                    _esaTitulo = value
                End Set
            End Property

			Private _esaCuerpo As String
            Public Property esaCuerpo() As String
                Get
					Return _esaCuerpo
                End Get
                Set(ByVal value As String)
                    _esaCuerpo = value
                End Set
            End Property

			Private _esaDescripcion As String
            Public Property esaDescripcion() As String
                Get
					Return _esaDescripcion
                End Get
                Set(ByVal value As String)
                    _esaDescripcion = value
                End Set
            End Property

			Private _esaActivo As Boolean
            Public Property esaActivo() As Boolean
                Get
					Return _esaActivo
                End Get
                Set(ByVal value As Boolean)
                    _esaActivo = value
                End Set
            End Property

			Private _esaFechaHoraCreacion As Date
            Public Property esaFechaHoraCreacion() As Date
                Get
					Return _esaFechaHoraCreacion
                End Get
                Set(ByVal value As Date)
                    _esaFechaHoraCreacion = value
                End Set
            End Property

			Private _esaFechaHoraUltimaModificacion As Date
            Public Property esaFechaHoraUltimaModificacion() As Date
                Get
					Return _esaFechaHoraUltimaModificacion
                End Get
                Set(ByVal value As Date)
                    _esaFechaHoraUltimaModificacion = value
                End Set
            End Property

			Private _cuoId As Integer
            Public Property cuoId() As Integer
                Get
					Return _cuoId
                End Get
                Set(ByVal value As Integer)
                    _cuoId = value
                End Set
            End Property

			Private _usuIdCreacion As Integer
            Public Property usuIdCreacion() As Integer
                Get
					Return _usuIdCreacion
                End Get
                Set(ByVal value As Integer)
                    _usuIdCreacion = value
                End Set
            End Property

			Private _usuIdUltimaModificacion As Integer
            Public Property usuIdUltimaModificacion() As Integer
                Get
					Return _usuIdUltimaModificacion
                End Get
                Set(ByVal value As Integer)
                    _usuIdUltimaModificacion = value
                End Set
            End Property

			Public Sub New()
                Try                
                    Me.esaId = 0
				Catch ex As Exception
                    Throw ex
                End Try
            End Sub            

			Public Sub New(ByVal esaId As Integer)
				Try
                    Fila = New DataTable
                    Fila = ocdGestor.EjecutarReader("[EmailSalida.ObtenerUno]", New Object(,) {{"@esaId", esaId}})

					If Fila.Rows.Count > 0 Then
						If Fila(0)("esaId").ToString().Trim().Length > 0 Then
                            Me.esaId = Convert.ToInt32(Fila(0)("esaId"))
                        Else
                            Me.esaId = 0
                        End If
                        
						If Fila(0)("esaFecha").ToString().Trim().Length > 0 Then
                            Me.esaFecha = Convert.ToDateTime(Fila(0)("esaFecha"))
                        Else
                            Me.esaFecha = Date.Now
                        End If
                        
						If Fila(0)("esaPara").ToString().Trim().Length > 0 Then
                            Me.esaPara = Convert.ToString(Fila(0)("esaPara"))
                        Else
                            Me.esaPara = ""
                        End If
                        
						If Fila(0)("esaTipo").ToString().Trim().Length > 0 Then
                            Me.esaTipo = Convert.ToString(Fila(0)("esaTipo"))
                        Else
                            Me.esaTipo = ""
                        End If
                        
						If Fila(0)("esaTitulo").ToString().Trim().Length > 0 Then
                            Me.esaTitulo = Convert.ToString(Fila(0)("esaTitulo"))
                        Else
                            Me.esaTitulo = ""
                        End If
                        
						If Fila(0)("esaCuerpo").ToString().Trim().Length > 0 Then
                            Me.esaCuerpo = Convert.ToString(Fila(0)("esaCuerpo"))
                        Else
                            Me.esaCuerpo = ""
                        End If
                        
						If Fila(0)("esaDescripcion").ToString().Trim().Length > 0 Then
                            Me.esaDescripcion = Convert.ToString(Fila(0)("esaDescripcion"))
                        Else
                            Me.esaDescripcion = ""
                        End If
                        
						If Fila(0)("esaFechaHoraCreacion").ToString().Trim().Length > 0 Then
                            Me.esaFechaHoraCreacion = Convert.ToDateTime(Fila(0)("esaFechaHoraCreacion"))
                        Else
                            Me.esaFechaHoraCreacion = Date.Now
                        End If
                        
						If Fila(0)("esaFechaHoraUltimaModificacion").ToString().Trim().Length > 0 Then
                            Me.esaFechaHoraUltimaModificacion = Convert.ToDateTime(Fila(0)("esaFechaHoraUltimaModificacion"))
                        Else
                            Me.esaFechaHoraUltimaModificacion = Date.Now
                        End If
                        
						If Fila(0)("esaActivo").ToString().Trim().Length > 0 Then
                            Me.esaActivo = IIf(Convert.ToInt32(Fila(0)("esaActivo")) = 1, True, False)
                        Else
							Me.esaActivo = False
                        End If
                        
						If Fila(0)("cuoId").ToString().Trim().Length > 0 Then
                            Me.cuoId = Convert.ToInt32(Fila(0)("cuoId"))
                        Else
							Me.cuoId = 0
                        End If
                        
						If Fila(0)("usuIdCreacion").ToString().Trim().Length > 0 Then
                            Me.usuIdCreacion = Convert.ToInt32(Fila(0)("usuIdCreacion"))
                        Else
							Me.usuIdCreacion = 0
                        End If
                        
						If Fila(0)("usuIdUltimaModificacion").ToString().Trim().Length > 0 Then
                            Me.usuIdUltimaModificacion = Convert.ToInt32(Fila(0)("usuIdUltimaModificacion"))
                        Else
							Me.usuIdUltimaModificacion = 0
                        End If
                        
					End If
				Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			Public Sub New(ByVal esaId As Integer, ByVal esaFecha As Date, ByVal esaPara As String, ByVal esaTipo As String, ByVal esaTitulo As String, ByVal esaCuerpo As String, ByVal esaDescripcion As String, ByVal esaActivo As Boolean, ByVal esaFechaHoraCreacion As Date, ByVal esaFechaHoraUltimaModificacion As Date, ByVal IcuoId As Integer, ByVal IusuIdCreacion As Integer, ByVal IusuIdUltimaModificacion As Integer)
				Try
					me.esaId = esaId
					me.esaFecha = esaFecha
					me.esaPara = esaPara
					me.esaTipo = esaTipo
					me.esaTitulo = esaTitulo
					me.esaCuerpo = esaCuerpo
					me.esaDescripcion = esaDescripcion
					me.esaActivo = esaActivo
					me.esaFechaHoraCreacion = esaFechaHoraCreacion
					me.esaFechaHoraUltimaModificacion = esaFechaHoraUltimaModificacion
					me.cuoId = cuoId
					me.usuIdCreacion = usuIdCreacion
					me.usuIdUltimaModificacion = usuIdUltimaModificacion
				Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			
			Public Function ObtenerParametros() As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[EmailSalida.ObtenerParametros]", New Object(,) {})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Function ObtenerTodo() As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[EmailSalida.ObtenerTodo]", New Object(,) {})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Function ObtenerTodoCustom() As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[EmailSalida.ObtenerTodoCustom]", New Object(,) {})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Function ObtenerUno(ByVal esaId as Integer) As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[EmailSalida.ObtenerUno]", New Object(,) {{"@esaId", esaId}})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Function ObtenerValidarParametro() As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[EmailSalida.ObtenerValidarParametro]", New Object(,) {})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Sub Actualizar(ByVal esaId as Integer, ByVal esaFecha as Date, ByVal esaPara as String, ByVal esaTipo as String, ByVal esaTitulo as String, ByVal esaCuerpo as String, ByVal esaDescripcion as String, ByVal cuoId as Integer, ByVal esaActivo as Boolean, ByVal usuIdCreacion as Integer, ByVal usuIdUltimaModificacion as Integer, ByVal esaFechaHoraCreacion as Date, ByVal esaFechaHoraUltimaModificacion as Date)
				Try
                    ocdGestor.EjecutarNonQuery("[EmailSalida.Actualizar]", New Object(,) {{"@esaId", esaId}, {"@esaFecha", esaFecha}, {"@esaPara", esaPara}, {"@esaTipo", esaTipo}, {"@esaTitulo", esaTitulo}, {"@esaCuerpo", esaCuerpo}, {"@esaDescripcion", esaDescripcion}, {"@cuoId", cuoId}, {"@esaActivo", esaActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@esaFechaHoraCreacion", esaFechaHoraCreacion}, {"@esaFechaHoraUltimaModificacion", esaFechaHoraUltimaModificacion}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub Eliminar(ByVal esaId as Integer)
				Try
                    ocdGestor.EjecutarNonQuery("[EmailSalida.Eliminar]", New Object(,) {{"@esaId", esaId}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub Generar(ByVal QueEnviar as String, ByVal Mes as Integer, ByVal Año as Integer, ByVal usuId as Integer)
				Try
                    ocdGestor.EjecutarNonQuery("[EmailSalida.Generar]", New Object(,) {{"@QueEnviar", QueEnviar}, {"@Mes", Mes}, {"@Año", Año}, {"@usuId", usuId}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub Insertar(ByVal esaFecha as Date, ByVal esaPara as String, ByVal esaTipo as String, ByVal esaTitulo as String, ByVal esaCuerpo as String, ByVal esaDescripcion as String, ByVal cuoId as Integer, ByVal esaActivo as Boolean, ByVal usuIdCreacion as Integer, ByVal usuIdUltimaModificacion as Integer, ByVal esaFechaHoraCreacion as Date, ByVal esaFechaHoraUltimaModificacion as Date)
				Try
                    ocdGestor.EjecutarNonQuery("[EmailSalida.Insertar]", New Object(,) {{"@esaFecha", esaFecha}, {"@esaPara", esaPara}, {"@esaTipo", esaTipo}, {"@esaTitulo", esaTitulo}, {"@esaCuerpo", esaCuerpo}, {"@esaDescripcion", esaDescripcion}, {"@cuoId", cuoId}, {"@esaActivo", esaActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@esaFechaHoraCreacion", esaFechaHoraCreacion}, {"@esaFechaHoraUltimaModificacion", esaFechaHoraUltimaModificacion}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub RegistrarEnvio(ByVal esaId as Integer)
				Try
                    ocdGestor.EjecutarNonQuery("[EmailSalida.RegistrarEnvio]", New Object(,) {{"@esaId", esaId}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub Actualizar()
                Try
                    If Me.esaId <> 0 Then
                        ocdGestor.EjecutarNonQuery("[EmailSalida.Actualizar]", New Object(,) {{"@esaId", esaId}, {"@esaFecha", esaFecha}, {"@esaPara", esaPara}, {"@esaTipo", esaTipo}, {"@esaTitulo", esaTitulo}, {"@esaCuerpo", esaCuerpo}, {"@esaDescripcion", esaDescripcion}, {"@cuoId", cuoId}, {"@esaActivo", esaActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@esaFechaHoraCreacion", esaFechaHoraCreacion}, {"@esaFechaHoraUltimaModificacion", esaFechaHoraUltimaModificacion}})
                    End If
                Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			Public Sub Eliminar()
                Try
                    If Me.esaId <> 0 Then
                        ocdGestor.EjecutarNonQuery("[EmailSalida.Eliminar]", New Object(,) {{"@esaId", esaId}})
                    End If
                Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			Public Function Insertar() as Integer
                Dim IdMax As Integer
                Try
                    IdMax = ocdGestor.EjecutarNonQueryRetornaId("[EmailSalida.Insertar]", New Object(,) {{"@esaFecha", esaFecha}, {"@esaPara", esaPara}, {"@esaTipo", esaTipo}, {"@esaTitulo", esaTitulo}, {"@esaCuerpo", esaCuerpo}, {"@esaDescripcion", esaDescripcion}, {"@cuoId", cuoId}, {"@esaActivo", esaActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@esaFechaHoraCreacion", esaFechaHoraCreacion}, {"@esaFechaHoraUltimaModificacion", esaFechaHoraUltimaModificacion}})
                Catch ex As Exception
                    Throw ex
                End Try
                Return IdMax
            End Function

		End Class
    End Namespace
End Namespace